base/src/test/kotlin/proguard/analysis/PartialEvaluatorErrorsTest.kt

```diff
@@ -0,0 +1,604 @@
+package proguard.analysis
+
+import io.kotest.assertions.throwables.shouldThrowAny
+import io.kotest.core.spec.style.FreeSpec
+import proguard.classfile.*
+import proguard.classfile.attribute.Attribute
+import proguard.classfile.attribute.CodeAttribute
+import proguard.classfile.attribute.visitor.AllAttributeVisitor
+import proguard.classfile.attribute.visitor.AttributeNameFilter
+import proguard.classfile.attribute.visitor.AttributeVisitor
+import proguard.classfile.editor.ClassBuilder
+import proguard.classfile.visitor.NamedMethodVisitor
+import proguard.evaluation.BasicInvocationUnit
+import proguard.evaluation.PartialEvaluator
+import proguard.evaluation.ParticularReferenceValueFactory
+import proguard.evaluation.value.DetailedArrayValueFactory
+import proguard.evaluation.value.ParticularValueFactory
+import proguard.testutils.AssemblerSource
+import proguard.testutils.ClassPoolBuilder
+
+
+/**
+ * The purpose of these tests is to find ProGuard Assembly snippets that will result in errors thrown by the
+ * `PartialEvaluator`.
+ *
+ * Some other issues have been discovered not directly related to the PartialEvaluator. Mainly issues when the code is
+ * first read.
+ *
+ * The logger should be able to figure out what the context is and provide context to the user that is debugging
+ * @see PartialEvaluator
+ */
+class PartialEvaluatorErrorsTest : FreeSpec({
+    val fastBuild = { impl: String ->
+        ClassPoolBuilder.fromSource(
+            AssemblerSource(
+                "PartialEvaluatorDummy.jbc",
+                """
+                public class PartialEvaluatorDummy extends java.lang.Object {
+                    $impl
+                }
+                """.trimIndent()
+            )
+        )
+    }
+
+    val fastEval =
+        { pool: ClassPool, partialEvaluator: PartialEvaluator, methodName: String, methodDescriptor: String ->
+            pool.classesAccept(
+                "PartialEvaluatorDummy", NamedMethodVisitor(
+                    methodName, methodDescriptor,
+                    AllAttributeVisitor(
+                        AttributeNameFilter(Attribute.CODE, partialEvaluator)
+                    )
+                )
+            )
+        }
+
+
+    /**
+     * This is a list of code snippets on which the `PartialEvaluator` throws on error
+     * but are in need of proper error messages with a concise and correct explanation of the error.
+     */
+    "Throws from partial evaluator but should be formatted" - {
```

Comment:
```suggestion
    "Throws from partial evaluator - {
```

For now, there is nothing about formatting here: these test are just testing that the PE throws exceptions (or not)